optional_plugins/runner_remote/tests/test_remote.py

```diff
@@ -99,7 +101,8 @@ def test_run_suite(self):
                         '0000000000000000000000000000000000000000 --json - '
                         '--archive /tests/sleeptest.py /tests/other/test '
                         'passtest.py --mux-yaml ~/avocado/tests/foo.yaml '
-                        '~/avocado/tests/bar/baz.yaml --dry-run')
+                        '~/avocado/tests/bar/baz.yaml --dry-run --filter-'
+                        'by-tags foo bar')
```

Comment:
This is actually not the way you'd use `--filter-by-tags`, it needs a single value.  You can pass it multiple times, though.  Which makes me think that the previous commit is `" ".join`ing them in the wrong way.

Reply:
I vaguely remember that there was a reason for that... the fact that listing tags together in one argument "AND"s them, and in separate arguments "OR"s them.  But, it looks like they would still work with `nargs='+' `.  Let me check and get back to you.

Reply:
Well this is the only one I know of that uses `append`, the rest of them use `nargs='+'`, so IMO it'd be better to use that. The only drawback is when someone used `avocado run --filter-by-tags quick passtest.py` which works currently but would consider `passtest.py` tag later. Anyway the same problem is there for `--mux-yaml` and other args...

Reply:
Sure, I'm all in favor of consistency, but let's coordinate and document the change of behavior instead of adding it to a unrelated fix.  Since avocado command lines out there may be broken by the change of behavior, and I don't see how to do a deprecation - so maybe a notice to the mailing list, followed by the change -- possibly on the next sprint.  How does it sound?

Reply:
Sure, I expected a deprecation procedure as I pointed out it might break some workflows... I'll do that and update this PR to work well.

Reply:
Thanks to Eduardo's email I remember why we used `append`. It allows specifying multiple expressions starting with `-`, which is (as far as I know) not possible with `nargs+`. Let's discuss on our meeting what to do with that, I'd still prefer unifying them which means turning --mux-yaml and others to `append`.